packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js

```diff
@@ -0,0 +1,237 @@
+// @ts-check
+import { assert, details } from '@agoric/assert';
+import { importBundle } from '@agoric/import-bundle';
+import { Remotable, getInterfaceOf, makeMarshal } from '@agoric/marshal';
+// grumble... waitUntilQuiescent is exported and closes over ambient authority
+import { waitUntilQuiescent } from '../../waitUntilQuiescent';
+
+import { makeLiveSlots } from '../liveSlots';
+
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+// eslint-disable-next-line no-unused-vars
+function workerLog(first, ...args) {
+  // @ts-ignore
+  // eslint-disable-next-line
+  // print(`---worker: ${first}`, ...args);
+}
+
+workerLog(`supervisor started`);
+
+/**
+ * @typedef { [unknown, ...unknown[]] } Tagged
+ */
+const Item = {
+  /** @type { (item: Tagged) => ArrayBuffer } */
+  encode: tagged => encoder.encode(JSON.stringify(tagged)).buffer,
+
+  /** @type { (msg: ArrayBuffer) => Tagged } */
+  decode(msg) {
+    const txt = decoder.decode(msg);
+
+    /** @type { Tagged } */
+    const item = JSON.parse(txt);
+    assert(Array.isArray(item), details`expected array`);
+    assert(item.length > 0, details`empty array lacks tag`);
+    return item;
+  },
+};
+
+function makeConsole(_tag) {
+  const log = console; // TODO: anylogger(tag);
+  const cons = {
+    debug: log.debug,
+    log: log.log,
+    info: log.info,
+    warn: log.warn,
+    error: log.error,
+  };
+  return harden(cons);
+}
+
+function testLog(...args) {
+  // @ts-ignore xsnap provides issueCommand global
+  // eslint-disable-next-line no-undef
+  issueCommand(Item.encode(['testLog', ...args]));
+}
+
+/**
+ * @param { (value: void) => void } f
+ * @param { string } errmsg
+ */
+function runAndWait(f, errmsg) {
+  Promise.resolve()
+    .then(f)
+    .then(undefined, err => workerLog(`doProcess: ${errmsg}:`, err));
+  return waitUntilQuiescent();
+}
+
+function makeWorker() {
+  /** @type { Record<string, (...args: unknown[]) => void> | null } */
+  let dispatch = null;
+
+  /** @type { (dr: Tagged, errmsg: string) => Promise<Tagged> } */
+  async function doProcess(dispatchRecord, errmsg) {
+    assert(dispatch);
+    const theDispatch = dispatch;
+    const [dispatchOp, ...dispatchArgs] = dispatchRecord;
+    assert(typeof dispatchOp === 'string');
```

Comment:
oops; this assert is vestigial from when I defined `Tagged` as `[string, ...unknown[]]` but there's never a need to know that the tag is a string without comparing it to a specific string value, so I switched to `[unknown, ...unknown[]]`.

Reply:
oops; I take that back; the `dispatchOp` _is_ used as an arbitrary string.

Reply:
Only in the case of the `dispatchOp` are the tag values not known at design-time -- they're method names. Changing the `Tagged` type to require the tag to be a `string` results in a bunch of noisy `assert`s with little value.